back-end/www/models/model_operations/location_operations.py

```diff
@@ -0,0 +1,367 @@
+        raise Exception(err_msg)
+
+    # Get the locations which have been answered by the user, or already done.
+    ex_location_filter = Location.query.filter(or_(Location.id.in_(user_answered_location_id_list), Location.done_at != None))
+    ex_location_id_list = [loc.id for loc in ex_location_filter.all()]
+
+    # Get locations which have gold standards
+    gold_location_filter = Location.query.filter(Location.id.in_(gold_location_id_list))
+
+    # Get to-be-excluded location ids, which are not either: 1. with gold answers 2. identified by the user before 3. Already labled done
+    if len(ex_location_id_list) > 0:
+        exclude_location_id_list = gold_location_id_list + ex_location_id_list
+    else:
+        exclude_location_id_list = gold_location_id_list
+
+    wait_test_locations_filter = Location.query.filter(Location.id.not_in(exclude_location_id_list))
+
+    sel_gold_location_list = []
+    sel_wait_test_locations_list = []
+
+    # Randomly sort and select the first locations which has been provideded gold answers
+    if gold_standard_size > 0:
+        rand_gold_location_list = gold_location_filter.order_by(func.random()).all()
+        dbprint("Got {} gold rand_gold_location_list.".format(len(rand_gold_location_list)))
+        sel_gold_location_list = rand_gold_location_list[0:gold_standard_size]
+        dbprint("sel_gold_location_list : ", sel_gold_location_list)
+
+    # Randomly sort and select the first locations which has exclude user answered ones plus those with gold answers
+    if size > gold_standard_size:
+        wait_test_locations_list = wait_test_locations_filter.order_by(func.random()).all()
+        sel_wait_test_locations_list = wait_test_locations_list[0:(size - gold_standard_size)]
+        dbprint("sel_wait_test_locations_list : ", sel_wait_test_locations_list)
+
+    # Combine and shuffle the list
+    location_list = sel_wait_test_locations_list
+    if len(sel_gold_location_list) > 0:
+        location_list += sel_gold_location_list
+        random.shuffle(location_list)
+
+    # Double confirm the amount
+    if len(location_list) < size:
+        raise Exception("Cannot find expected amount of locations", size)
+
+    return location_list
+
+
+def get_location_is_done_count():
+    """
+    Get the count of locations which have been labled done.
+
+    Returns
+    -------
+    count : int
+        The amount of locations which have been labled done.
+    """
+    # Create an exclusive filter to get locations which have done_at date
+    location_query = Location.query.filter(Location.done_at.isnot(None))
+    count = location_query.count()
+    return count
+
+
+def get_location_count():
+    """
+    Get the total number of locations.
+
+    Returns
+    -------
+    count : int
+        The total number of locations.
+    """
+    count = Location.query.count()
+    return count
+
+
+def batch_process_answers(user_id, answers):
```

Comment:
Should this function be in the answer_operations.py file? (Since this function will be called in the "/answer/" API, not the "/location/" API, right?)

Reply:
I have trouble putting it in answer_operations.py due to complex dependency between answer and location. In C we usually do this with a header file, but I have not identified how to do this with Python....any suggestions?